### PR TITLE
fix(flutter/a11y assessments): h1 missing a11y from each page on the web app

### PR DESCRIPTION
**Pull request by essjay05**
Adds Semantic Widget to wrap the contents of each page's AppBar title attribute and adds header: true and headingLevel: 1 as Semantic attributes so the content of the title is compiled as an h1 tag to meet accessibility guidelines that each page must have a h1 tag present. Also updates the test cases for the home page and each use case page to check for its respective h1.

[Before Screenshot - Accessibility Assessments Home Page](https://screenshot.googleplex.com/4i9LuiGwvLnEcZ8)
[Before Screenshot - Checkbox List Title -- note: each use case page is missing an h1](https://screenshot.googleplex.com/3qQjfqvAMTehRsm)
[After Screenshot - Accessibility Assessments Home Page](https://screenshot.googleplex.com/APSJJXBmwNBP35m)
[After Screenshot - Checkbox List Title-- note: change is similar across each Accessibility use case page](https://screenshot.googleplex.com/6EGgZnTusEgeN5L)

Fixes b/338035526

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: dev/a11y_assessments/lib/main.dart [hunk 1]
```diff
@@ -81,7 +81,13 @@ class HomePageState extends State<HomePage> {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(title: const Text('Accessibility Assessments')),
+      appBar: AppBar(
```
Comment (chunhtai) [inline dev/a11y_assessments/lib/main.dart:84]: Instead of changing all example, should we directly bake headinglevel into Appbar widget?

Comment (essjay05) [reply dev/a11y_assessments/lib/main.dart:84]: @chunhtai is [this change\(https://github.com/flutter/flutter/pull/152198/commits/3b9f9b260681ddd69740a651a789f7d6058889a3) sufficient for this?

Comment (essjay05) [reply dev/a11y_assessments/lib/main.dart:84]: Was attempting to fix the a11y_assessments separately and then attempt to do the AppBar with `headingLevel: 1` baked in after? Open to whatever you think is best @chunhtai.

Comment (chunhtai) [reply dev/a11y_assessments/lib/main.dart:84]: we should do https://github.com/flutter/flutter/pull/152198/commits/3b9f9b260681ddd69740a651a789f7d6058889a3 instead

Comment (essjay05) [reply dev/a11y_assessments/lib/main.dart:84]: Updated all use_case widgets to revert to previous and rely on app_bar default `headingLevel: 1`, all new tests still passed.

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -991,6 +991,9 @@ class _AppBarState extends State<AppBar> {
             TargetPlatform.iOS || TargetPlatform.macOS => null,
           },
           header: true,
+          // set default headingLevel: 1 for one h1 element
+          // per page to meet accessibility guidelines
+          headingLevel: 1,
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/app_bar.dart:996]: do we need this change to pass gar? I thought if we convert header to banner, It should have worked?

Comment (essjay05) [reply packages/flutter/lib/src/material/app_bar.dart:996]: I believe so, the [bug](b/338035526) I'm trying to fix references [GAR 1.6](go/GAR-web#overview-of-criterion-16) where it is required to have at least one h1 tag on each page. 

Let me know if this is a good way to enforce this rule.

Comment (essjay05) [reply packages/flutter/lib/src/material/app_bar.dart:996]: Attempted to make related engine changes for [issue](https://github.com/flutter/flutter/issues/152268) in [this PR](https://github.com/flutter/engine/pull/54224).

Comment (chunhtai) [reply packages/flutter/lib/src/material/app_bar.dart:996]: This seems not addressed yet

**Review by chunhtai — COMMENTED** [review]
waiting for review comment to be resolved

**Comment by essjay05** [comment]
> waiting for review comment to be resolved

Just updated to remove default `headingLevel: 1` in this commit: https://github.com/flutter/flutter/pull/152198/commits/1dc9aba75a390675cd92a5258e5908c46dc2dd78.

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by hannah-hyj — APPROVED** [review]
LGTM

**Pull request merged**